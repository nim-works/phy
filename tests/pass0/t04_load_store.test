discard """
  output: "(Done 100)"
"""
(TypeDefs
  (Int 8)
  (ProcTy (Type 0))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Locals)
    (Continuations
      (Continuation (Params) 8
        (Store (Type 0) (Addr (IntVal 0)) (IntVal 100))
        (Continue 1 (Load (Type 0) (Addr (IntVal 0))))
      )
      (Continuation (Params (Type 0)))
    )
  )
)