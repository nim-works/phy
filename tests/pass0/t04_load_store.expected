.type t0 (Int)
.type t1 (Proc t0)
.start t1 p0
.local lo0 int
  StackAlloc 8
  PopLocal lo0
  GetLocal lo0
  LdImmInt 100
  WrInt64 0
  GetLocal lo0
  LdInt64 0
  Ret
.end
